File: python/examples/async/basic.py
import asyncio

from mirascope import llm


@llm.call("openai/gpt-5-mini")
async def recommend_book(genre: str):
    return f"Recommend a {genre} book."


async def main():
    response = await recommend_book("fantasy")
    print(response.text())


asyncio.run(main())

File: typescript/examples/prompts/basic.ts
/**
 * Basic prompt usage example.
 *
 * Demonstrates using definePrompt to create a reusable prompt
 * that can be called with different models.
 *
 * Run with: bun run example examples/prompts/basic.ts
 */
import { llm } from "mirascope";

/**
 * A reusable prompt for recommending books.
 * The prompt is model-agnostic - you choose the model at call time.
 */
const recommendBook = llm.definePrompt<{ genre: string }>({
  template: ({ genre }) => `Please recommend a book in ${genre}.`,
});

// Call the prompt with a model
const response = await recommendBook("anthropic/claude-haiku-4-5", {
  genre: "fantasy",
});

console.log(response.text());
// The Name of the Wind by Patrick Rothfuss...
